tests/election.py

```diff
@@ -111,9 +131,14 @@ def run(args):
 
             try:
                 test_kill_primary(network, args)
-            except PrimaryNotFound:
+            except (PrimaryNotFound, TimeoutError) as e:
                 if node_to_stop < nodes_to_stop - 1:
                     raise
+                elif not (
+                    (type(e) == PrimaryNotFound and args.consensus == "cft")
+                    or (type(e) == TimeoutError and args.consensus == "bft")
```

Comment:
I don't understand why this distinction is necessary or useful here. test_kill_primary should not exhibit consensus-specific behaviour here, only nodes_to_stop (calculated accordingly) should be allowed to differ?